### PR TITLE
.github: Introduce Github-centric config helper files

### PR DESCRIPTION
**Pull request by nderjung**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://docs.unikraft.org/contribute.html) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [x] Updated relevant documentation.

### Base target

 - Architecture(s): N/A.
 - Platform(s): N/A.
 - Application(s): N/A.

### Additional configuration

N/A.

### Description of changes

This series introduces a few new files which aid in the use of Github as a central repository management system for the Unikraft OSS project.  Particularly, this series introduces:

 * `.github/CODEOWNERS` which protects directories and files and aids with distributing assignments of incoming PRs;
 * `.github/ISSUE_TEMPLATE/bug_report.yml` which provides [an issue template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates#issue-templates) for reporting a bug;
 * `.github/ISSUE_TEMPLATE/project_backlog.yml` which provides an issue template for recommending a new feature;
 *  `.github/ISSUE_TEMPLATE/config.yml` which adds an additional link to the "new issue" page to direct feature requests to the [discussions forum](https://github.com/unikraft/unikraft/discussions) and,
 * `.github/PULL_REQUEST_TEMPLATE.md` which provides a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates#pull-request-templates) for the body of text when a new PR is opened.

_N.B. This PR description is formatted using the proposed template._

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# https://help.github.com/articles/about-codeowners/
+
+# These owners are the default owners for all Unikraft core files.
+*                     @unikraft/maintainers-core
+
+# Along with all default owners, these owners oversee hardware architectures.
+/arch/                @unikraft/maintainers-arch
```
Comment (razvand) [inline .github/CODEOWNERS:7]: There aren't teams named `maintainers-arch`, `maintainers-plat`, `maintainers-libs` yet.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+        Please only use this form to submit problems with builds or the runtime of Unikraft unikernels.  This includes unexpected or undefined behavior, stacktraces, critical messages, and things just breaking or exploding.
+        
+        **Ensure you have tested against the latest changes on `staging` branch before submitting this form.**
+
+        For questions, concerns or help with running Unikraft, please check out our additional resources:
+
+        * [Unikraft Documentation](https://docs.unikraft.org)
+        * [Getting Started Guide](https://unikraft.org/getting-started)
+        * [Unikraft Discussions Forum](https://github.com/unikraft/unikraft/discussions)
```
Comment (razvand) [inline .github/ISSUE_TEMPLATE/bug_report.yml:18]: Should we still use the forum? Or should we ditch it in favor of Discord?

**Review by razvand — COMMENTED** [review]
I have to comments, the rest looks good to me.

File: .github/CODEOWNERS (hunk 1 above)
Comment (nderjung) [reply .github/CODEOWNERS:7]: Hi @razvand,

These lists will be populated automatically by a "governance" repository defining who is in these lists via a YAML file.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (nderjung) [reply .github/ISSUE_TEMPLATE/bug_report.yml:18]: We should add Discord too and keep for the forum as a public Q&A, FAQ and for RFCs.

**Review by razvand — APPROVED** [review]
Reviewed-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>

**Pull request closed without merging**